### ISSUE TITLE
IndexedVar domain display not updated

### ISSUE THREAD

**Issue by qtothec** [Pyomo/pyomo#5]
When the domain of a Var is not preset and its elements are modified, the default domain of Reals is displayed even when all elements are changed consistently to a different domain. Small example below:

```
"""
Small example showing outdated Var domain display
"""

from pyomo.environ import ConcreteModel, Var, RangeSet, Binary

m = ConcreteModel()
m.s = RangeSet(2)
m.x = Var(m.s)
m.x[1].domain = Binary
m.x[2].domain = Binary

m.x.display()

"""Result is
x : Size=2, Index=s, Domain=Reals
    Key : Lower : Value : Upper : Fixed : Stale
      1 :     0 :  None :     1 : False :  True
      2 :     0 :  None :     1 : False :  True
despite both individual vars having a domain of Binary
"""
```

**Comment by ghackebeil** [comment]
Thanks. Good catch. I’ve fixed this on trunk.

Gabe

**Comment by ghackebeil** [comment]
fixed in 238b867c05e58ae331b8df03a91b181a9938c876
